unified-runtime/source/adapters/opencl/event.cpp

```diff
@@ -181,7 +181,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventGetInfo(ur_event_handle_t hEvent,
     size_t CheckPropSize = 0;
     cl_int RetErr = clGetEventInfo(hEvent->CLEvent, CLEventInfo, propSize,
                                    pPropValue, &CheckPropSize);
-    if (pPropValue && CheckPropSize != propSize) {
+    if (pPropValue && CheckPropSize != propSize &&
+        propName != UR_EVENT_INFO_COMMAND_EXECUTION_STATUS) {
+      // Opencl:cpu may (incorrectly) return 0 for propSize when checking
+      // execution status when status is CL_COMPLETE.
```

Comment:
This change seems unrelated to the rest of the patch, it also seems more like a driver bug to me. Is fixing this in the OpenCL adapter really the best direction?

Reply:
It is not intended to a be a long term solution. Just temporary until the driver is fixed and bumped.   My understanding is driver fixes take longer and I really want to get these teardown issues addressed before we worsen them ( again ).